src/codewhisperer/models/constants.ts

```diff
@@ -127,6 +127,7 @@ export const CodeWhispererConstants = {
     codeScanZipExt: '.zip',
     contextTruncationTimeoutSeconds: 10,
     codeScanJobTimeoutSeconds: 50,
+    projectSizeCalculateTimeoutSeconds: 10,
```

Comment:
10s seems high. Especially just for telemetry. Does this risk a bad experience for customers?

Reply:
CodeScan telemetry is now async and decoupled from the feature running progress, set to 10s in case edge case which customer's project is very big; Previous job's telemetry will be guaranteed to emit before coming job finish as the SLA is 60s, average cost is ~30s now

Reply:
TBH, I don't think we really need a timeout for this. It's just a telemetry and the UX is not blocked on this (or at least it should not be).

Reply:
No, we need, have done some benchmark, even system command can take minutes to finish if it's too large; timeout is def better experience and prevents eating up customer's system resources

Reply:
> and the UX is not blocked on this (or at least it should not be).

Couldn't it block other Toolkit features, or greatly slow them down? This is only one case also; there are many such recursive workspace functions in codewhisperer, each adding risk. A timeout or some sort of guard is something we should have in each case as a general habit. The minimum would be to log a warning, if not cancel the operation.

Reply:
Totally agree, that's why for all those recursion in CWSPR, we have wrap with timeout either on time or size